src/vm/database/sqlite.rs

```diff
@@ -0,0 +1,335 @@
+        })
+    }
+
+    fn load_contract(&self, contract_name: &str) -> Result<Option<Contract>> {
+        let contract: Option<String> =
+            self.query_row(
+                "SELECT contract_data FROM contracts WHERE contract_name = ?",
+                &[contract_name],
+                |row| {
+                    row.get(0)
+                });
+        match contract {
+            None => Ok(None),
+            Some(ref contract) => Ok(Some(
+                Contract::deserialize(contract)
+                    .expect(DESERIALIZE_FAIL_MESSAGE)))
+        }
+    }
+
+    pub fn create_map(&mut self, contract_name: &str, map_name: &str, key_type: TupleTypeSignature, value_type: TupleTypeSignature) -> Result<()> {
+        let key_type = TypeSignature::new_atom(AtomTypeIdentifier::TupleType(key_type));
+        let value_type = TypeSignature::new_atom(AtomTypeIdentifier::TupleType(value_type));
+
+        self.execute("INSERT INTO maps_table (contract_name, map_name, key_type, value_type) VALUES (?, ?, ?, ?)",
+                     &[contract_name, map_name, &key_type.serialize()?, &value_type.serialize()?]);
+        Ok(())
+    }
+
+    pub fn fetch_entry(&self, contract_name: &str, map_name: &str, key: &Value) -> Result<Value> {
+        let map_descriptor = self.load_map(contract_name, map_name)?;
+        if !map_descriptor.key_type.admits(key) {
+            return Err(Error::new(ErrType::TypeError(format!("{:?}", map_descriptor.key_type), (*key).clone())))
+        }
+
+        let params: [&ToSql; 2] = [&map_descriptor.map_identifier,
+                                   &key.serialize()?];
+
+        let sql_result: Option<Option<String>> = 
+            self.query_row(
+                "SELECT value FROM data_table WHERE map_identifier = ? AND key = ? ORDER BY data_identifier DESC LIMIT 1",
+                &params,
+                |row| {
+                    row.get(0)
+                });
+        match sql_result {
+            None => {
+                Ok(Value::Void)
+            },
+            Some(sql_result) => {
+                match sql_result {
+                    None => Ok(Value::Void),
+                    Some(value_data) => Ok(Value::deserialize(&value_data)
+                                           .expect(DESERIALIZE_FAIL_MESSAGE))
+                }
+            }
+        }
+    }
+
+    pub fn set_entry(&mut self, contract_name: &str, map_name: &str, key: Value, value: Value) -> Result<Value> {
+        let map_descriptor = self.load_map(contract_name, map_name)?;
+        if !map_descriptor.key_type.admits(&key) {
+            return Err(Error::new(ErrType::TypeError(format!("{:?}", map_descriptor.key_type), key)))
+        }
+        if !map_descriptor.value_type.admits(&value) {
+            return Err(Error::new(ErrType::TypeError(format!("{:?}", map_descriptor.value_type), value)))
+        }
+
+        let params: [&ToSql; 3] = [&map_descriptor.map_identifier,
```

Comment:
What happens if `.serialize()` fails?  Do we want to `panic!` or `unreachable!` here, or is this recoverable?

Reply:
These should panic